src/helpers/universalRouter.js

```diff
@@ -22,12 +22,16 @@ const fetchDataForContainers = (containers, store, params, query) => {
 export default function universalRouter(location, history, store, preload) {
   const routes = createRoutes(history);
   return new Promise((resolve, reject) => {
-    match({routes, history, location}, (error, redirectLocation, renderProps) => {
+    const matchCallback = (error, redirectLocation, renderProps) => {
       if (error) {
         return reject(error);
       }
 
       if (redirectLocation) {
+        if (history) {
+          history.replaceState(null, redirectLocation.pathname);
+          return match({routes, history, location: redirectLocation}, matchCallback);
+        }
         return resolve({
```

Comment:
I had an `if/else` block here, but eslint yelled saying "no else after return". I don't like that rule, but it's a bare `if` statement now.